recommender/services/fts.py

```diff
@@ -44,6 +45,33 @@ def filter_services(search_data: SearchData) -> QuerySet:
 
 
 def retrieve_services_for_recommendation(
+    elastic_services: List[int], accessed_services: Optional[Iterable] = None
+) -> QuerySet:
+    """
+    Selecting candidates for recommendation
+
+    Args:
+        elastic_services: Marketplace's context
+        accessed_services: Services that user accessed
+    """
+
+    q = deepcopy(elastic_services)
```

Comment:
Maybe this time we would also want to have a comment explaining the memory leak as the reason for deepcopy?

Reply:
deepcopy here was not necessary

Reply:
deepcopy here was not necessary